azurelinuxagent/ga/update.py

```diff
@@ -670,6 +671,8 @@ def _report_status(self, exthandlers_handler):
             if self._goal_state is not None:
                 agent_status = exthandlers_handler.get_ext_handlers_status_debug_info(vm_status)
                 self._goal_state.save_to_history(agent_status, AGENT_STATUS_FILE)
+                if self._goal_state.extensions_goal_state.is_outdated:
+                    exthandlers_handler.protocol.client.get_host_plugin().clear_fast_track_state()
```

Comment:
we cleared state but status report had failures due to network issues, so we continue to skip following goal states until CRP receives not supported feature?

Reply:
exthandlers_handler.report_ext_handlers_status() should return None on error, and then we wouldn't clear the state.

But the point you make is valid: CRP will continue sending FastTrack goal states until the agent can report status successfully. That is a requirement of the current protocol.

Those goal states would time out. Something similar would happen (regardless of FastTrack) if the agent cannot report status. CRP needs the status to know the goal state completed. One difference, though, is that in this scenario the agent would not even see the FastTrack goal states.

Reply:
I see your point, in that case VM might not be in good state (whole different problem) and should not worry about Agent behavior, I guess.